vault_cli/cli.py

```diff
@@ -367,11 +367,21 @@ def delete(client_obj: client.VaultClientBase, name: str, key: Optional[str]) ->
     required=True,
     help="Folder or single item. Pass several times to load multiple values. You can use --path mypath=prefix or --path mypath:key=prefix if you want to change the generated names of the environment variables",
 )
+@click.option(
+    "-o",
+    "--omit-single-key",
+    is_flag=True,
```

Comment:
If you write the long form as `--omit-single-key/--no-omit-single-key`, you'll get an explicit way of disabling the behaviour should it have been switched by `VAULT_CLI_ENV_OMIT_SINGLE_KEY` and click will undestand it's a flag automatically. Also,:
>  Click always wants you to provide an enable and disable flag so that you can change the default later.

https://click.palletsprojects.com/en/7.x/options/#boolean-flags